src/main/java/gregtech/api/capability/impl/fecompat/GTEnergyItemWrapper.java

```diff
@@ -0,0 +1,104 @@
+package gregtech.api.capability.impl.fecompat;
+
+import gregtech.api.capability.IElectricItem;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.energy.CapabilityEnergy;
+import net.minecraftforge.energy.IEnergyStorage;
+
+import java.util.function.BiConsumer;
+
+public class GTEnergyItemWrapper implements IElectricItem {
+
+    /**
+     * Capability Provider of the FE TileEntity for the EU-capability.
+     */
+    private final ICapabilityProvider upvalue;
+
+    /**
+     * Capability holder for the FE-capability.
+     */
+    private IEnergyStorage energyStorage;
+
+    protected GTEnergyItemWrapper(ICapabilityProvider upvalue) {
+        this.upvalue = upvalue;
+    }
+
+    /**
+     * Test this EnergyContainer for sided Capabilities.
+     *
+     * @param side The side of the TileEntity to test for the Capability
+     * @return True if side has Capability, false otherwise
+     */
+    protected boolean isValid(EnumFacing side) {
+        return upvalue.hasCapability(CapabilityEnergy.ENERGY, side);
+    }
+
+    private IEnergyStorage getEnergyStorage() {
+        if(energyStorage == null) {
+            energyStorage = upvalue.getCapability(CapabilityEnergy.ENERGY, null);
+        }
+        return energyStorage;
+    }
+
+    @Override
+    public boolean canProvideChargeExternally() {
+        return true;
+    }
+
+    @Override
+    public boolean chargeable() {
+        IEnergyStorage storage = getEnergyStorage();
+        if(storage != null)
+            return storage.canReceive();
```

Comment:
Should this also check if the storage is not at its maximum stored amount? aka: `storage.getEnergyStored == storage.getMaxEnergyStored`